x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/ProfilingIndexTemplateRegistry.java

```diff
@@ -46,7 +46,8 @@ public class ProfilingIndexTemplateRegistry extends IndexTemplateRegistry {
     // version 3: Add optional component template 'profiling-ilm@custom' to all ILM-managed index templates
     // version 4: Added 'service.name' keyword mapping to profiling-events
     // version 5: Add optional component template '<idx-name>@custom' to all index templates that reference component templates
-    public static final int INDEX_TEMPLATE_VERSION = 5;
+    // version 6: Added 'host.arch' keyword mapping to profiling-hosts
+    public static final int INDEX_TEMPLATE_VERSION = 6;
```

Comment:
To be clear: this will *not* force a rollover of the current backing index of `profiling-hosts`, i.e. the mapping changes will apply only after that index rolls over regularly (which can be days after the upgrade). If that's ok, then you can leave this as is, otherwise if you want to force rollover during the upgrade to 8.14.0 you need to bump `PROFILING_HOSTS_VERSION` as well.

Reply:
Thanks for pointing out. This was also mentioned in the comments further down (thanks for those).
We keep both fields and mappings but will write only to the new field (host agent + collector are about to be changed right now).

Btw, are there tracking issues of things to be done for 9.0.0? It would be nice to add all these little things...

Reply:
> Btw, are there tracking issues of things to be done for 9.0.0? It would be nice to add all these little things...

I've created https://github.com/elastic/elasticsearch/issues/106592 as a meta-issue so we can collect all related issues. Can you please create a new issue for that and then add it as a list item to the meta issue?

Reply:
@rockdaboot Shouldn't we bump PROFILING_HOSTS_VERSION here as well? If we don't, then we could have upgraded agents in an upgraded cluster sending `host.arch` (but not `profiling.host.machine`) which ends up getting dropped as the mapping doesn't exist yet. The result is no architecture information at all!